packages/erxes-ui/src/components/richTextEditor/utils/getParsedMentions.tsx

```diff
@@ -1,10 +1,25 @@
 export const getParsedMentions = (data: any) => {
-  const mentions = (data.content || []).flatMap(getParsedMentions);
-  if (data.type === 'mention') {
-    mentions.push(data.attrs.id);
-  }
+  const mentionIds = new Set();
 
-  const uniqueMentions = [...new Set(mentions)] as any;
+  // Recursive function to traverse nested content
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **issue (complexity):** Consider simplifying the updated implementation to reduce complexity while still handling nested content.

While the updated implementation successfully introduces recursion to handle nested content, which is a solid approach for traversing structures like the one you're dealing with, it does significantly increase the complexity of the code. The original version had a straightforward, linear flow that was easy to follow, leveraging `flatMap` and a `Set` to collect unique mention IDs. The new version, with its recursive `traverseContent` function and additional checks, introduces a higher cognitive load for understanding and maintaining the code.

Here's a suggestion to maintain the simplicity of the original approach while incorporating the necessary functionality to handle nested content. This version focuses on the core requirement of collecting unique mention IDs and simplifies the condition checks and the way mentions are collected:

```javascript
export const getParsedMentions = (data: any) => {
  const mentionIds = new Set();

  const traverseContent = (content: any) => {
    (content || []).forEach((item: any) => {
      if (item.type === 'mention') {
        mentionIds.add(item.attrs.id);
      } else if (Array.isArray(item.content)) {
        traverseContent(item.content);
      }
    });
  };

  traverseContent(data.content);

  return Array.from(mentionIds);
};
```

This approach keeps the recursive traversal to handle nested structures but streamlines the process, making the code easier to read and maintain. It directly targets items of type 'mention' to add their IDs to the `Set`, extending the original code's intent to handle nested structures without the added complexity of the new implementation.